File: app/src/main/java/com/example/achalker/starbuzz/DrinkCategoryActivity.java
package com.example.achalker.starbuzz;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

public class DrinkCategoryActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_drink_category);

        ArrayAdapter<Drink> listAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, Drink.drinks);

        ListView listDrinks = (ListView) findViewById(R.id.list_drinks);
        listDrinks.setAdapter(listAdapter);

        AdapterView.OnItemClickListener itemClickListener = new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> listDrinks, View itemView, int position, long id) {

                Intent intent = new Intent(DrinkCategoryActivity.this, DrinkActivity.class);
                intent.putExtra(DrinkActivity.EXTRA_DRINKED, (int) id);
                startActivity(intent);
            }
        };

        listDrinks.setOnItemClickListener(itemClickListener);
    }
}

File: app/src/main/java/com/example/achalker/starbuzz/DrinkCategoryActivity.kt
package com.example.achalker.starbuzz

import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.widget.AdapterView.OnItemClickListener
import android.widget.ArrayAdapter
import android.widget.ListView
import com.example.achalker.starbuzz.DrinkActivity
import com.example.achalker.starbuzz.model.Drink

class DrinkCategoryActivity : AppCompatActivity() {
    companion object {
        private const val EXTRA_DRINK_NAME = "EXTRA_DRINK_NAME"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_drink_category)

        val drinkNames = Drink.drinks.map { it.name }
        val listAdapter = ArrayAdapter(this, R.layout.list_item, drinkNames)

        val listDrinks = findViewById<View>(R.id.list_drinks) as ListView
        listDrinks.adapter = listAdapter

        val itemClickListener = OnItemClickListener { _, _, position, _ ->
            val intent = Intent(this@DrinkCategoryActivity, DrinkActivity::class.java)
            intent.putExtra(EXTRA_DRINK_NAME, Drink.drinks[position].name)
            startActivity(intent)
        }
        listDrinks.onItemClickListener = itemClickListener
    }
}